lib/task/jobs/jobWorkerTask.class.php

```
@@ -101,6 +102,27 @@ public function getPhpReportedMemoryUsage()
         return sprintf('%.0f', memory_get_usage(true) / 1024);
     }
 
+    public function getMaxMemUsage()
+    {
+        return $this->maxMemUsage;
+    }
+
+    public function setMaxMemUsage(int $maxMemUsage)
+    {
+        if ($maxMemUsage > 0) {
```

Nitpick: this defaults to 0 so maybe it should be settable to 0? Or maybe it should throw an error, rather than fail silently, if it's given the value 0?

---

I think I'm going to leave this - the default value of 0 equates to the setting being not in use. Allowing it to be set to zero has no benefit. If the value set is greater than zero, a message will be displayed in the log on atom-worker startup indicating 'Worker will shut down if memory consumption exceeds %ukB.' - this message will be absent if the setting is set to a nonsense value: 0 or negative. This mirrors what was done for `--max-jobs-count`.  In most cases this will be set once in `/usr/lib/systemd/system/atom-worker.service` and left.